### PR TITLE
HTCONDOR-2358 Add condor_q -jobset option

### PR DESCRIPTION
**Pull request by johnkn**
to query and display jobse ads.

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_q.V6/queue.cpp [hunk 1]
```diff
@@ -3815,15 +3820,15 @@ show_schedd_queue(const char* scheddAddress, const char* scheddName, const char*
...
 	}
-	if ((useFastPath > 1) && ((fetch_opts & CondorQ::fetch_FromMask) == CondorQ::fetch_Jobs)) {
+	if ((useFastPath > 1) && ((fetch_opts & QueryFetchOpts::fetch_FromMask) == QueryFetchOpts::fetch_Jobs)) {
 		if (dash_tot && ! dash_unmatchable) {
-			fetch_opts |= CondorQ::fetch_SummaryOnly;
+			fetch_opts |= fetch_SummaryOnly;
 #ifdef CONDOR_Q_HANDLE_CLUSTER_AD 
 		} else if (dash_factory && (dash_long || ! dash_batch)) {
-			fetch_opts |= CondorQ::fetch_IncludeClusterAd;
+			fetch_opts |= fetch_IncludeClusterAd;
```
Comment (ColeBollig) [inline src/condor_q.V6/queue.cpp:3828]:
```suggestion
			fetch_opts |= QueryFetchOpts::fetch_IncludeClusterAd;
```

File: src/condor_q.V6/queue.cpp [hunk 2]
```diff
@@ -3815,15 +3820,15 @@ show_schedd_queue(const char* scheddAddress, const char* scheddName, const char*
 	if ( ! fetch_opts && (useFastPath > 1)) {
 		fetch_opts = default_fetch_opts;
 	}
-	if ((useFastPath > 1) && ((fetch_opts & CondorQ::fetch_FromMask) == CondorQ::fetch_Jobs)) {
+	if ((useFastPath > 1) && ((fetch_opts & QueryFetchOpts::fetch_FromMask) == QueryFetchOpts::fetch_Jobs)) {
 		if (dash_tot && ! dash_unmatchable) {
-			fetch_opts |= CondorQ::fetch_SummaryOnly;
+			fetch_opts |= fetch_SummaryOnly;
```
Comment (ColeBollig) [inline src/condor_q.V6/queue.cpp:3825]:
```suggestion
			fetch_opts |= QueryFetchOpts::fetch_SummaryOnly;
```

File: docs/version-history/feature-versions-23-x.rst [hunk 3]
```diff
@@ -51,6 +51,10 @@ New Features:
   with that singularity or apptainer container runtime.
   :jira:`2180`
 
+- A new ``-jobset`` display option was added to *condor_q*.  When jobsets are enabled
```
Comment (ColeBollig) [inline docs/version-history/feature-versions-23-x.rst:54]:
```suggestion
- A new ``-jobset`` display option was added to :tool:`condor_q`.  When jobsets are enabled
```

File: docs/man-pages/condor_q.rst [hunk 4]
```diff
@@ -416,7 +416,9 @@ Options
     normal output of *condor_q* will be blank.
  **-factory**
     (output option) Display information about late materialization job
-    factories in the *condor_shedd*. 
+    factories in the *condor_shedd*.
```
Comment (olifre) [inline docs/man-pages/condor_q.rst:419]: `condor_schedd` :wink:

File: docs/man-pages/condor_q.rst [hunk 5]
```diff
@@ -416,7 +416,9 @@ Options
...
+ **-jobset**
+    (output option) Display information about jobsets in the *condor_shedd*
```
Comment (olifre) [inline docs/man-pages/condor_q.rst:421]: See above :wink: .

File: docs/man-pages/condor_q.rst (hunk 4 above)
Comment (ColeBollig) [reply docs/man-pages/condor_q.rst:419]: Good catch.

**Review by ColeBollig — APPROVED** [review]
LGTM

**Pull request merged**